### DISCUSSION TITLE
What is the raven-database-topology files?

### DISCUSSION THREAD

**Discussion by mfonnesbeck** [ravendb/ravendb#16983] [Q&A] [answered]
We recently upgraded our databases and application code from 3.X to 5.X and noticed that connecting to a database keeps creating these *.raven-database-topology extensioned files. What are they, how are they used, and should we add them to source control? I cannot seem to find any mention of them anywhere in the documentation or online.

**Comment by ppekrol** [comment, marked as answer]
Hi,

from v4.x client caches the topology in case all the servers provided in the DocumentStore.Urls are down. By doing that, we can still load the topology, and try to connect to nodes that are not in the list of Urls. This can be disabled by DocumentStore.Conventions.DisableTopologyCache if needed.

Hope that this helps.
